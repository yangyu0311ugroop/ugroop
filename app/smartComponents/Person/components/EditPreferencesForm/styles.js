const styles = {
  root: {},
  grow: {
    flex: '1',
  },
  popper: {
    marginTop: 4,
    zIndex: 9999,
    fontWeight: 400,
    width: 300,
  },
  popperContainer: {
    padding: 16,
    width: 300,
  },
  seeMore: {
    marginTop: 10,
  },
  fullWidth: {
    width: '100%',
  },
  formWidth: {
    padding: '0 16px',
  },
  live: {
    color: 'white',
    fontWeight: 500,
    position: 'relative',
    background: '#fff',
    borderRadius: '50%',
    fontSize: 10,

    '&:before': {
      animation: 'pulsate1 1.2s ease-out',
      animationIterationCount: 'infinite',
      border: '6px solid green',
      borderRadius: 5,
      content: "' '",
      left: -14,
      position: 'absolute',
      top: 8,
    },
  },
  reminderText: {
    textTransform: 'none',
  },
  offsetGrid: {
    padding: 8,
  },
};

export default styles;
