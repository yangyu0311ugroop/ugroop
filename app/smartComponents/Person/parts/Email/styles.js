const styles = {
  root: {},
  grow: {
    flex: '1',
  },
  inviteContainer: {
    paddingRight: '0px !important',
  },
  inviteButton: {
    paddingLeft: '0px',
    paddingRight: '0px',
    paddingTop: '4px',
    fontStyle: 'italic',
    fontSize: '12px',
  },
  actionButton: {
    border: 'unset',
    color: 'unset',
    minHeight: 'unset',
    padding: '2px 4px',
    background: 'unset',
    boxShadow: 'unset',

    '&:hover': {
      background: '#c5c5c54f',
    },
  },
  darkButton: {
    color: 'white',
  },
  popper: {
    marginTop: 4,
    zIndex: 9999,
    fontWeight: 400,
    width: 300,
  },
  popperContainer: {
    padding: 16,
    width: 300,
  },
  emailEllipsis: {
    maxWidth: 190,
    float: 'left',
  },
  noUnderline: {
    borderBottom: 'none !important',
    cursor: 'inherit !important',
    textDecoration: 'none !important',
  },
  icon: {
    marginRight: 10,
    /* fontSize: 20,
    lineHeight: '.9', */
  },
  noWrap: {
    whiteSpace: 'nowrap',
  },
};

export default styles;
