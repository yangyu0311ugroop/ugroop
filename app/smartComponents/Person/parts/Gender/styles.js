const styles = {
  root: {},
  radioGroup: {
    flexDirection: 'row',
  },
  formLabel: {
    fontSize: 10,
    fontWeight: 500,
  },
  grow: {
    flex: '1',
  },
};

export default styles;
