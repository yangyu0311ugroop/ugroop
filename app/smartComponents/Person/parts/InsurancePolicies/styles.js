const styles = {
  root: {},
  grow: {
    flex: '1',
  },
  emptyCard: {
    padding: 0,
  },
};

export default styles;
