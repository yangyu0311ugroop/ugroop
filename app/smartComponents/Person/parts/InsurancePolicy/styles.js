const styles = ({ colors }) => ({
  lineIndicator: {
    width: 40,
    height: 2,
    backgroundColor: colors.base,
  },
  name: {
    lineHeight: 1.6,
    marginTop: 8,
  },
  noWordBreak: {
    wordBreak: 'normal',
  },
});

export default styles;
