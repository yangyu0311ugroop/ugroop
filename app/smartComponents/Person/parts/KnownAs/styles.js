const styles = ({ colors }) => ({
  lineIndicator: {
    width: 40,
    height: 2,
    backgroundColor: colors.base,
  },
  name: {
    marginTop: 8,
  },
  noWordBreak: {
    wordBreak: 'normal',
  },
});

export default styles;
