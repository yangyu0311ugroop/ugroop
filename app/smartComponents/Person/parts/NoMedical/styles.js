const styles = {
  root: {},
  grow: {
    flex: '1',
  },
  checkBoxRoot: {
    height: 24,
    width: 30,
    color: 'blue',
  },
  checked: {
    color: 'black !important',
  },
};

export default styles;
