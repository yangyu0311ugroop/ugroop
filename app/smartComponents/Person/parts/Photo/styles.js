const styles = {
  root: {
    borderRadius: '50%',
    width: 120,
    height: 120,
  },
  medium: {
    width: 120,
    height: 120,
  },
  round: {
    borderRadius: '50%',
  },
  square: {
    borderRadius: 4,
  },
  grow: {
    flex: '1',
  },
};

export default styles;
