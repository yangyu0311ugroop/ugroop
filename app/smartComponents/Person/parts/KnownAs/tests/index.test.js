import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import mockStylesheet from 'utils/mockStylesheet';
import theme from 'theme/coolTheme';
import Form from 'ugcomponents/Form';
import { TEST_HELPERS } from 'utils/helpers/testHelpers';
import { VARIANTS } from 'variantsConstants';

import styles from '../styles';
import { KnownAs } from '../index';

const mockStyles = mockStylesheet('KnownAs', styles, theme);

describe('<KnownAs />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const intl = {
    formatMessage: jest.fn(),
  };

  const props = {
    classes: mockStyles,
    resaga,
    intl,
  };

  beforeEach(() => {
    rendered = shallow(<KnownAs {...props} />);
    instance = rendered.instance();
    jest.clearAllMocks();
  });

  it('should exists', () => {
    expect(KnownAs).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('componentDidMount', () => {
    it('should setValue if there is no hasPersonDetail and there is organisationId', () => {
      rendered.setProps({
        hasPersonDetail: false,
        organisationId: 1,
      });
      instance.componentDidMount();
      expect(resaga.setValue).toBeCalled();
    });
  });
  describe('getEditableName()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.getEditableName()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
  describe('handleEditableSubmit()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(
        <div>
          {instance.handleEditableSubmit({
            model: {},
            onSuccess: jest.fn(),
            onError: jest.fn(),
          })}
        </div>,
      );

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
  describe('renderStringOnly', () => {
    it('should render tooltip', () => {
      rendered.setProps({
        isTooltip: true,
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderStringOnly);
    });
    it('should render isPersonal', () => {
      rendered.setProps({
        isPersonal: true,
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderStringOnly);
    });
    it('should render default', () => {
      rendered.setProps({
        isPersonal: false,
        isTooltip: false,
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderStringOnly);
    });
  });
  describe('renderEllipsisString()', () => {
    it('should render correctly', () => {
      rendered.setProps({
        maxCharCountToEllipsis: 30,
        firstName: 'ccc',
        lastName: 'ddd',
      });
      const snapshot = shallow(<div>{instance.renderEllipsisString()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render correctly if name length is greater than maxCharCountToEllipsis', () => {
      rendered.setProps({
        maxCharCountToEllipsis: 1,
        knownAs: 'QQQAAA',
      });
      const snapshot = shallow(<div>{instance.renderEllipsisString()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });

  describe('renderValueOnly', () => {
    it('should return knownAs if knownAs have value', () => {
      rendered.setProps({
        knownAs: 'Ye',
      });
      const result = instance.renderValueOnly();

      expect(result).toBe('Ye');
    });

    it('should return firstname + lastname if knownAs is null', () => {
      rendered.setProps({
        firstName: 'AA',
        lastName: 'BB',
        knownAs: null,
      });
      const result = instance.renderValueOnly();

      expect(result).toBe('AA BB');
    });
  });

  describe('renderTextOnly', () => {
    it('should render tooltip if isMatchToName is true', () => {
      rendered.setProps({
        isMatchToName: true,
        knownAs: 'known As',
        firstName: '',
        lastName: 'last name',
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderTextOnly);
    });

    it('should render tooltip if isMatchToName is true should show alias', () => {
      rendered.setProps({
        isMatchToName: true,
        knownAs: 'john L',
        firstName: 'john',
        lastName: 'L',
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderTextOnly);
    });

    it('should render tooltip if isMatchToName is false', () => {
      rendered.setProps({
        isMatchToName: false,
      });
      TEST_HELPERS.expectMatchSnapshot(instance.renderTextOnly);
    });
  });

  describe('renderEditable()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.renderEditable()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text only variant', () => {
      rendered.setProps({
        variant: VARIANTS.TEXT_ONLY,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text field if variant is text field', () => {
      rendered.setProps({
        variant: VARIANTS.TEXT_FIELD,
      });
      const snapshot = shallow(<Form>{instance.render()}</Form>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text field if variant is string only', () => {
      rendered.setProps({
        variant: VARIANTS.STRING_ONLY,
      });
      const snapshot = shallow(<Form>{instance.render()}</Form>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
