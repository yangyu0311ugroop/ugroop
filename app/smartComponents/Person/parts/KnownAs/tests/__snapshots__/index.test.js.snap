// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KnownAs /> getEditableName() should render correctly 1`] = `
<div>
  person.knownAs
</div>
`;

exports[`<KnownAs /> handleEditableSubmit() should render correctly 1`] = `<div />`;

exports[`<KnownAs /> render() should render correctly 1`] = `
<div>
  <div
    className="KnownAs-lineIndicator"
  />
  <H1>
    <div
      className="j-text-ellipsis"
      title=" "
    >
       
    </div>
  </H1>
</div>
`;

exports[`<KnownAs /> render() should render text field if variant is string only 1`] = `
<Fragment>
  <ContextProvider
    value={
      Object {
        "isFormDisabled": [Function],
      }
    }
  >
    <Formsy
      disabled={false}
      getErrorMessage={[Function]}
      getErrorMessages={[Function]}
      getValue={[Function]}
      hasValue={[Function]}
      isFormDisabled={[Function]}
      isFormSubmitted={[Function]}
      isPristine={[Function]}
      isRequired={[Function]}
      isValid={[Function]}
      isValidValue={[Function]}
      mapping={null}
      onChange={[Function]}
      onError={[Function]}
      onInvalid={[Function]}
      onInvalidSubmit={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
      onValid={[Function]}
      onValidSubmit={[Function]}
      preventExternalInvalidation={false}
      resetValue={[Function]}
      setValidations={[Function]}
      setValue={[Function]}
      showError={[Function]}
      showRequired={[Function]}
      validationErrors={null}
    >
      <Wrapped
        canSubmitForm={true}
        cancelButtonContent="Discard"
        cancelForm={[Function]}
        isFormChanged={false}
        isFormDisabled={false}
        loading={false}
        renderCancelButton={[Function]}
        renderSubmitButton={[Function]}
        resetForm={[Function]}
        submitButtonContent="Submit"
        submitForm={[Function]}
      >
        <div
          className="j-text-ellipsis"
          title=" "
        >
           
        </div>
      </Wrapped>
    </Formsy>
  </ContextProvider>
  <DiscardConfirmationDialog
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`<KnownAs /> render() should render text field if variant is text field 1`] = `
<Fragment>
  <ContextProvider
    value={
      Object {
        "isFormDisabled": [Function],
      }
    }
  >
    <Formsy
      disabled={false}
      getErrorMessage={[Function]}
      getErrorMessages={[Function]}
      getValue={[Function]}
      hasValue={[Function]}
      isFormDisabled={[Function]}
      isFormSubmitted={[Function]}
      isPristine={[Function]}
      isRequired={[Function]}
      isValid={[Function]}
      isValidValue={[Function]}
      mapping={null}
      onChange={[Function]}
      onError={[Function]}
      onInvalid={[Function]}
      onInvalidSubmit={[Function]}
      onReset={[Function]}
      onSubmit={[Function]}
      onValid={[Function]}
      onValidSubmit={[Function]}
      preventExternalInvalidation={false}
      resetValue={[Function]}
      setValidations={[Function]}
      setValue={[Function]}
      showError={[Function]}
      showRequired={[Function]}
      validationErrors={null}
    >
      <Wrapped
        canSubmitForm={true}
        cancelButtonContent="Discard"
        cancelForm={[Function]}
        isFormChanged={false}
        isFormDisabled={false}
        loading={false}
        renderCancelButton={[Function]}
        renderSubmitButton={[Function]}
        resetForm={[Function]}
        submitButtonContent="Submit"
        submitForm={[Function]}
      >
        <FormsyInterceptor
          breakWord={true}
          component={[Function]}
          ellipsis={false}
          intl={
            Object {
              "formatMessage": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "defaultMessage": "Known As",
                      "id": "app.smartComponents.Person.parts.KnownAs.label",
                    },
                  ],
                  Array [
                    Object {
                      "defaultMessage": "Known As",
                      "id": "app.smartComponents.Person.parts.KnownAs.label",
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
            }
          }
          isToolTip={false}
          maxCharCountToEllipsis={32}
          name="knownAs"
          onInterceptValue={[Function]}
          value=""
        />
      </Wrapped>
    </Formsy>
  </ContextProvider>
  <DiscardConfirmationDialog
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  />
</Fragment>
`;

exports[`<KnownAs /> render() should render text only variant 1`] = `
<div>
  <P
    breakWord={true}
    className=""
    component={[Function]}
    ellipsis={false}
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
    isToolTip={false}
    maxCharCountToEllipsis={32}
  >
    <div
      className="j-text-ellipsis"
      title=" "
    >
       
    </div>
     
  </P>
</div>
`;

exports[`<KnownAs /> renderEditable() should render correctly 1`] = `
<div>
  <WithStyles(EditableTextForm)
    label={
      <FormattedMessage
        defaultMessage="Known As"
        id="app.smartComponents.Person.parts.KnownAs.label"
      />
    }
    name="person.knownAs"
    onSubmit={[Function]}
    placeholder="Click to specify preferred name"
    readOnly={false}
    required={true}
    value=""
  />
</div>
`;

exports[`<KnownAs /> renderEllipsisString() should render correctly 1`] = `
<div>
  <span
    title="ccc ddd"
  >
    ccc ddd
  </span>
</div>
`;

exports[`<KnownAs /> renderEllipsisString() should render correctly if name length is greater than maxCharCountToEllipsis 1`] = `
<div>
  <span
    className="j-text-ellipsis"
  >
    QQQAAA
  </span>
</div>
`;

exports[`<KnownAs /> renderStringOnly should render default 1`] = `
<div>
  <div
    className="j-text-ellipsis"
    title=" "
  >
     
  </div>
</div>
`;

exports[`<KnownAs /> renderStringOnly should render isPersonal 1`] = `
<div>
  <div
    className="j-text-ellipsis"
    title=" "
  >
     
  </div>
</div>
`;

exports[`<KnownAs /> renderStringOnly should render tooltip 1`] = `
<div>
  <div
    className="j-text-ellipsis"
    title=" "
  >
     
  </div>
</div>
`;

exports[`<KnownAs /> renderTextOnly should render tooltip if isMatchToName is false 1`] = `
<div>
  <P
    breakWord={true}
    className=""
    component={[Function]}
    ellipsis={false}
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
    isMatchToName={false}
    isToolTip={false}
    maxCharCountToEllipsis={32}
  >
    <div
      className="j-text-ellipsis"
      title=" "
    >
       
    </div>
     
  </P>
</div>
`;

exports[`<KnownAs /> renderTextOnly should render tooltip if isMatchToName is true 1`] = `
<div>
  <P
    breakWord={true}
    className=""
    component={[Function]}
    ellipsis={false}
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
    isMatchToName={true}
    isToolTip={false}
    maxCharCountToEllipsis={32}
  >
    <WithStyles(JTextClass)
      bold={true}
      md={true}
    >
      <div
        className="j-text-ellipsis"
        title="known As"
      >
        ( last name)
      </div>
       
    </WithStyles(JTextClass)>
  </P>
</div>
`;

exports[`<KnownAs /> renderTextOnly should render tooltip if isMatchToName is true should show alias 1`] = `
<div>
  <P
    breakWord={true}
    className=""
    component={[Function]}
    ellipsis={false}
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
    isMatchToName={true}
    isToolTip={false}
    maxCharCountToEllipsis={32}
  />
</div>
`;
